### DISCUSSION TITLE
KV cache APIs for LLM inference

### DISCUSSION THREAD

**Discussion by prashantaithal** [microsoft/onnxruntime#21589] [API Q&A] [answered]
Does ORT either provide any APIs in python or C++  or an  implementation of kv cache that would help optimize LLM inference? 
If no, is there any guide or reference on how the kv cache can be implemented in ORT?

**Comment by carzh** [comment, marked as answer]
[ORT GenAI](https://github.com/microsoft/onnxruntime-genai) provides a higher-level API that handles the KV cache
